### DISCUSSION TITLE
Custom Colors - How does Mantine resolve custom color css variables?

### DISCUSSION THREAD

**Discussion by slugmandrew** [mantinedev/mantine#6161] [Q&A]
Hi,

I am currently setting up a project with custom colors on V7. I have added my custom colors to the theme and everything seems to work well, except... my IDE does not seem to recognise the css variables generated to reference the custom colors. I am using IntelliJ:

![image](https://github.com/mantinedev/mantine/assets/153625/bdf745a1-72d6-4ac0-90e8-bb2e71e4d92d)

I know the variable is there, as the color is set correctly, so it seems to be an IDE issue. I was wondering if anyone had come across this problem or knew of any way to possibly fix it by making the IDE aware of the vars?

Thanks

**Comment by rtivital** [comment]
These variables are defined in runtime and not included in css files. JetBrains IDEs are configured to scan for declared variables in css files only. To fix an error, you can create a css file that defines those variables or just turn off css validation.

**Reply by slugmandrew** [reply]
Thanks for the super fast reply!

So just so I'm clear, the solution is to define the vars manually in a root css file, and then manually keep them in sync with the theme object?

Could vanilla extract help me here, or does it solve a different problem?

**Comment by matthjes** [comment]
If this question is still relevant: I fixed this by marking the directory `node_modules/@mantine` as a "Generated sources" type of directory.

**Reply by Andre741** [reply]
Hi, how did you do this? I can't find the "Generated sources" type of directory
Thanks
